File: pipeline/problems/b3e639a5-d2b6-4ce0-9d38-1c0777b2a3ba/Outputs/CodeContentFiles/NodeJS/solution.js
class Solution {
  static summarizeDiagonalEchoes(gridData) {
    const m = gridData.length;
    const n = m > 0 ? gridData[0].length : 0;
    const diagonals = new Map();

    for (let r = 0; r < m; r++) {
      if (r % 2 === 0) {
        for (let c = 0; c < n; c++) {
          const key = r + c;
          if (!diagonals.has(key)) diagonals.set(key, []);
          diagonals.get(key).push(gridData[r][c]);
        }
      } else {
        for (let c = n - 1; c >= 0; c--) {
          const key = r + c;
          if (!diagonals.has(key)) diagonals.set(key, []);
          diagonals.get(key).push(gridData[r][c]);
        }
      }
    }

    const keys = Array.from(diagonals.keys()).sort((a, b) => a - b);
    const result = [];

    for (let i = 0; i < keys.length; i++) {
      const d = keys[i];
      const group = diagonals.get(d).slice().sort((a, b) => a - b);
      const size = group.length;

      if (size % 2 === 1) {
        const median = group[Math.floor(size / 2)];
        result.push(Number(median).toFixed(2));
      } else {
        let total = 0;
        for (let j = 0; j < size; j++) total += group[j];
        const avg = total / size;
        result.push(avg.toFixed(2));
      }
    }

    return result.join(" ");
  }
}

File: pipeline/problems/b3e639a5-d2b6-4ce0-9d38-1c0777b2a3ba/Outputs/CodeContentFiles/Cpp/solution.cpp
#include <bits/stdc++.h>
using namespace std;

class solution {
public:
    string summarizeDiagonalEchoes(vector<vector<int>>& gridData) {
        int m = (int)gridData.size();
        int n = m > 0 ? (int)gridData[0].size() : 0;
        map<int, vector<int> > diagonals;

        for (int r = 0; r < m; r++) {
            if (r % 2 == 0) {
                for (int c = 0; c < n; c++) {
                    diagonals[r + c].push_back(gridData[r][c]);
                }
            } else {
                for (int c = n - 1; c >= 0; c--) {
                    diagonals[r + c].push_back(gridData[r][c]);
                }
            }
        }

        vector<string> result;
        for (map<int, vector<int> >::iterator it = diagonals.begin(); it != diagonals.end(); ++it) {
            vector<int> group = it->second;
            sort(group.begin(), group.end());
            int size = (int)group.size();

            ostringstream oss;
            oss << fixed << setprecision(2);

            if (size % 2 == 1) {
                double median = group[size / 2];
                oss << median;
            } else {
                long long s = 0;
                for (int i = 0; i < size; i++) s += group[i];
                double avg = (double)s / (double)size;
                oss << avg;
            }
            result.push_back(oss.str());
        }

        string out = "";
        for (size_t i = 0; i < result.size(); i++) {
            out += result[i];
            if (i + 1 < result.size()) out += " ";
        }
        return out;
    }
};